src/gui/widgets/text_box_base.cpp

```diff
@@ -138,31 +138,18 @@ void text_box_base::set_cursor(const std::size_t offset, const bool select)
 	reset_cursor_state();
 
 	if(select) {
-
-		if(selection_start_ == offset) {
-			selection_length_ = 0;
-		} else {
-			selection_length_ = -static_cast<int>(selection_start_ - offset);
-		}
-
+		selection_length_ = (selection_start_ == offset) ? 0 : -static_cast<int>(selection_start_ - offset);
 #ifdef __unix__
 		// selecting copies on UNIX systems.
 		copy_selection(true);
```

Comment:
Still think this should be dropped. It's weird behavior...

Reply:
removed. Also found more redundancy in clipboard.cpp, and decided to remove that as well. See https://github.com/wesnoth/wesnoth/pull/9356/commits/60a4361aebdb66fe3b622b290ea79a41b73c2aee.

Reply:
SDL2 doesn't support multi clipboards anyway, plus there's X11 vs Wayland behavior differences.